crates/runtime/src/catalogconnector/glue/provider.rs

```diff
@@ -0,0 +1,475 @@
+/*
+Copyright 2024-2025 The Spice.ai OSS Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+     https://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+use super::state::GlueCatalogState;
+use super::{Error, Result, TableType};
+use crate::dataconnector::DataConnectorFactory as _;
+use crate::dataconnector::parameters::aws::load_config;
+use crate::{
+    Runtime,
+    component::{catalog::Catalog, dataset::builder::DatasetBuilder},
+    dataconnector::{parameters::ConnectorParams, s3::S3Factory},
+};
+use async_trait::async_trait;
+use aws_sdk_glue::{Client, types::Table};
+use aws_sdk_sts::config::ProvideCredentials as _;
+use data_components::RefreshableCatalogProvider;
+use datafusion::{
+    catalog::{CatalogProvider, SchemaProvider, TableProvider},
+    common::Result as DFResult,
+    error::DataFusionError,
+};
+use iceberg::{
+    NamespaceIdent, TableIdent,
+    io::{FileIOBuilder, S3_ACCESS_KEY_ID, S3_SECRET_ACCESS_KEY},
+    spec::TableMetadata,
+    table::Table as IcebergTable,
+};
+use iceberg_datafusion::IcebergTableProvider;
+use snafu::prelude::*;
+use std::sync::Arc;
+use std::{any::Any, collections::HashMap, fmt};
+
+/// A catalog provider for AWS Glue, managing databases and tables.
+pub struct GlueCatalogProvider {
+    pub(super) inner: Arc<GlueCatalogState>,
+}
+
+impl fmt::Debug for GlueCatalogProvider {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        f.debug_struct("GlueCatalogProvider")
+            .finish_non_exhaustive()
+    }
+}
+
+/// A schema provider for a specific Glue database, providing table metadata.
+pub struct GlueSchemaProvider {
+    database: String,
+    state: Arc<GlueCatalogState>,
+}
+
+impl fmt::Debug for GlueSchemaProvider {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        f.debug_struct("GlueSchemaProvider")
+            .field("database", &self.database)
+            .finish_non_exhaustive()
+    }
+}
+
+impl GlueCatalogProvider {
+    pub async fn new(
+        mut parameters: ConnectorParams,
+        catalog: &Catalog,
+        runtime: Arc<Runtime>,
+    ) -> Result<Self> {
+        for validator in super::VALIDATORS.iter() {
+            validator
+                .validate(&mut parameters)
+                .await
+                .context(super::ParameterValidationSnafu)?;
+        }
+
+        parameters
+            .parameters
+            .insert("file_format".to_string(), "parquet".into());
```

Comment:
This isn't a limitation of AWS glue, but rather us constraining ourselves to hive style parquet, correct?